test/Microsoft.Extensions.Configuration.UserSecrets.Test/UserSecretsTestFixture.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.Extensions.Configuration.UserSecrets.Test
+{
+    public class UserSecretsTestFixture : IDisposable
+    {
+        public const string TestSecretsId = "d6076a6d3ab24c00b2511f10a56c68cc";
+
+        private Stack<Action> _disposables = new Stack<Action>();
+
+        public UserSecretsTestFixture()
+        {
+            _disposables.Push(() => TryDelete(Path.GetDirectoryName(PathHelper.GetSecretsPathFromSecretsId(TestSecretsId))));
+        }
+
+        public void Dispose()
+        {
+            while (_disposables.Count > 0)
+            {
+                _disposables.Pop()?.Invoke();
```

Comment:
In theory lots of things can happen 😄 Why would it ever be null in any implementation? Presumably we'd only push stuff if there's an actual action to push? It's of course safe to leave it the way it is, though.